### ISSUE TITLE
maximum session exceeded on osx after login

### ISSUE THREAD

**Issue by dylanfoster** [pingidentity/mod_auth_openidc#287]
###### Environment

- mod_auth_openidc version: 2.3.1
- Apache version 2.4.25
- platform/distro macOS 10.12.6 (16G29)

###### Expected behaviour
after login user is redirected to target site

###### Actual behaviour
after login user is redirected to google auth again

###### Minimized example
*Minimal, complete configuration that reproduces the behavior. Use the mailing list or get commercial support to discuss your own (full) setup.*

```
OIDCProviderMetadataURL https://accounts.google.com/.well-known/openid-configuration
OIDCClientID <>
OIDCClientSecret <>
OIDCRedirectURL <site root>/oauth_callback.html
OIDCCryptoPassphrase <>
OIDCScope "openid email profile"

<Directory "/Library/WebServer/Documents">
  AllowOverride ALL
  AuthType openid-connect
  Require valid-user
</Directory>
```

###### Configuration and Log files

```
oidc_check_max_session_duration: maxium session duration exceeded for user <user>

**Comment by zandbelt** [comment]
Can you attach server debug logs or send those over directly?

**Comment by dylanfoster** [comment]
```
[Sat Aug 05 18:58:19.675476 2017] [auth_openidc:debug] [pid 52464] src/util.c(616): [client 73.158.230.14:63889] oidc_util_http_form_encoded_data: data=grant_type=authorization_code&code=<code>&redirect_uri=<site_root>/oauth_callback.html&state=<state>&client_id=<client_token>&client_secret=<client_secret>
[Sat Aug 05 18:58:19.675484 2017] [auth_openidc:debug] [pid 52464] src/util.c(640): [client 73.158.230.14:63889] oidc_util_http_call: url=https://www.googleapis.com/oauth2/v4/token, data=grant_type=authorization_code&code=<code>&redirect_uri=<site_root>/oauth_callback.html&state=<state>&client_id=<client_id>&client_secret=<client_secret>, content_type=application/x-www-form-urlencoded, basic_auth=(null), bearer_token=(null), ssl_validate_server=1, timeout=60, outgoing_proxy=(null), pass_cookies=0, ssl_cert=(null), ssl_key=(null)
[Sat Aug 05 18:58:19.832637 2017] [auth_openidc:debug] [pid 52464] src/util.c(777): [client 73.158.230.14:63889] oidc_util_http_call: HTTP response code=200
[Sat Aug 05 18:58:19.832658 2017] [auth_openidc:debug] [pid 52464] src/util.c(782): [client 73.158.230.14:63889] oidc_util_http_call: response={\n "access_token": "<token>",\n "token_type": "Bearer",\n "expires_in": 3600,\n "id_token": "<id_token>"\n}\n
[Sat Aug 05 18:58:19.832852 2017] [auth_openidc:debug] [pid 52464] src/proto.c(2529): [client 73.158.230.14:63889] oidc_proto_validate_code_response: enter
[Sat Aug 05 18:58:19.832891 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1553): [client 73.158.230.14:63889] oidc_proto_parse_idtoken: enter: id_token header={"alg":"RS256","kid":"<kid>"}
[Sat Aug 05 18:58:19.832906 2017] [auth_openidc:debug] [pid 52464] src/util.c(1957): [client 73.158.230.14:63889] oidc_util_create_symmetric_key: key_len=32
[Sat Aug 05 18:58:19.832958 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1575): [client 73.158.230.14:63889] oidc_proto_parse_idtoken: successfully parsed (and possibly decrypted) JWT with header={"alg":"RS256","kid":"<kid>"}, and payload={"azp":"<azp>","aud":"<aud>","sub":"<sub>","hd":"<domain>","email":"<email>","email_verified":true,"at_hash":"<hash>","nonce":"<nonce>","iss":"https://accounts.google.com","iat":1501973899,"exp":1501977499,"name":"Dylan Foster","picture":"","given_name":"Dylan","family_name":"Foster","locale":"en"}
[Sat Aug 05 18:58:19.832970 2017] [auth_openidc:debug] [pid 52464] src/util.c(1957): [client 73.158.230.14:63889] oidc_util_create_symmetric_key: key_len=24
[Sat Aug 05 18:58:19.832980 2017] [auth_openidc:debug] [pid 52464] src/metadata.c(633): [client 73.158.230.14:63889] oidc_metadata_jwks_get: enter, jwks_uri=https://www.googleapis.com/oauth2/v3/certs, refresh=0
[Sat Aug 05 18:58:19.832987 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(536): [client 73.158.230.14:63889] oidc_cache_get: enter: https://www.googleapis.com/oauth2/v3/certs (section=j, decrypt=0, type=shm)
[Sat Aug 05 18:58:19.832997 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(570): [client 73.158.230.14:63889] oidc_cache_get: cache hit: return 1472 bytes from shm cache backend for key https://www.googleapis.com/oauth2/v3/certs
[Sat Aug 05 18:58:19.833039 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1334): [client 73.158.230.14:63889] oidc_proto_get_key_from_jwks: search for kid "<kid>" or thumbprint x5t "(null)"
[Sat Aug 05 18:58:19.833105 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1396): [client 73.158.230.14:63889] oidc_proto_get_key_from_jwks: found matching kid: "<kid>" for jwk: {"kty":"RSA","kid":"<kid>","e":"AQAB","n":"<n>"}
[Sat Aug 05 18:58:19.833126 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1465): [client 73.158.230.14:63889] oidc_proto_get_keys_from_jwks_uri: returning 1 key(s) obtained from the (possibly cached) JWKs URI
[Sat Aug 05 18:58:19.833202 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1510): [client 73.158.230.14:63889] oidc_proto_jwt_verify: JWT signature verification with algorithm "RS256" was successful
[Sat Aug 05 18:58:19.833212 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1286): [client 73.158.230.14:63889] oidc_proto_validate_idtoken: enter, jwt.header="{"alg":"RS256","kid":"<kid>"}", jwt.payload={"azp":"<azp>","aud":"<aud>","sub":"<sub>","hd":"<domain>","email":"<email>","email_verified":true,"at_hash":"<hash>","nonce":"<nonce>","iss":"https://accounts.google.com","iat":1501973899,"exp":1501977499,"name":"Dylan Foster","picture":"","given_name":"Dylan","family_name":"Foster","locale":"en"}", nonce=<nonce>
[Sat Aug 05 18:58:19.833222 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(536): [client 73.158.230.14:63889] oidc_cache_get: enter: qNz_dm7X346gEm2HEbkP-9gzZAc9yTs4c-JYQw0f-Io (section=n, decrypt=0, type=shm)
[Sat Aug 05 18:58:19.833265 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(572): [client 73.158.230.14:63889] oidc_cache_get: cache miss from shm cache backend for key qNz_dm7X346gEm2HEbkP-9gzZAc9yTs4c-JYQw0f-Io
[Sat Aug 05 18:58:19.833272 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(595): [client 73.158.230.14:63889] oidc_cache_set: enter: qNz_dm7X346gEm2HEbkP-9gzZAc9yTs4c-JYQw0f-Io (section=n, len=43, encrypt=0, ttl(s)=1210, type=shm)
[Sat Aug 05 18:58:19.833285 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(622): [client 73.158.230.14:63889] oidc_cache_set: successfully stored 43 bytes in shm cache backend for key qNz_dm7X346gEm2HEbkP-9gzZAc9yTs4c-JYQw0f-Io
[Sat Aug 05 18:58:19.833290 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1001): [client 73.158.230.14:63889] oidc_proto_validate_nonce: nonce "<nonce>" validated successfully and is now cached for 1210 seconds
[Sat Aug 05 18:58:19.833298 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1097): [client 73.158.230.14:63889] oidc_proto_validate_cnf: enter: policy=optional
[Sat Aug 05 18:58:19.833302 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1105): [client 73.158.230.14:63889] oidc_proto_validate_cnf: no Provided Token Binding ID environment variable found
[Sat Aug 05 18:58:19.833308 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1614): [client 73.158.230.14:63889] oidc_proto_parse_idtoken: valid id_token for user "<token>" expires: [Sat, 05 Aug 2017 23:58:19 GMT], in 3600 secs from now)
[Sat Aug 05 18:58:19.833320 2017] [auth_openidc:debug] [pid 52464] src/proto.c(2389): [client 73.158.230.14:63889] oidc_proto_validate_hash: hash_len=16, decoded_len=16, calc_len=32
[Sat Aug 05 18:58:19.833325 2017] [auth_openidc:debug] [pid 52464] src/proto.c(2402): [client 73.158.230.14:63889] oidc_proto_validate_hash: successfully validated the provided "at_hash" hash value (<value>) against the calculated value
[Sat Aug 05 18:58:19.833330 2017] [auth_openidc:debug] [pid 52464] src/mod_auth_openidc.c(1124): [client 73.158.230.14:63889] oidc_retrieve_claims_from_userinfo_endpoint: enter
[Sat Aug 05 18:58:19.833341 2017] [auth_openidc:debug] [pid 52464] src/proto.c(2160): [client 73.158.230.14:63889] oidc_proto_resolve_userinfo: enter, endpoint=https://www.googleapis.com/oauth2/v3/userinfo, access_token=<token>
[Sat Aug 05 18:58:19.833347 2017] [auth_openidc:debug] [pid 52464] src/util.c(599): [client 73.158.230.14:63889] oidc_util_http_query_encoded_url: url=https://www.googleapis.com/oauth2/v3/userinfo
[Sat Aug 05 18:58:19.833352 2017] [auth_openidc:debug] [pid 52464] src/util.c(640): [client 73.158.230.14:63889] oidc_util_http_call: url=https://www.googleapis.com/oauth2/v3/userinfo, data=(null), content_type=(null), basic_auth=(null), bearer_token=<token>, ssl_validate_server=1, timeout=60, outgoing_proxy=(null), pass_cookies=0, ssl_cert=(null), ssl_key=(null)
[Sat Aug 05 18:58:19.973634 2017] [auth_openidc:debug] [pid 52464] src/util.c(777): [client 73.158.230.14:63889] oidc_util_http_call: HTTP response code=200
[Sat Aug 05 18:58:19.973651 2017] [auth_openidc:debug] [pid 52464] src/util.c(782): [client 73.158.230.14:63889] oidc_util_http_call: response={\n "sub": "<sub>",\n "name": "Dylan Foster",\n "given_name": "Dylan",\n "family_name": "Foster",\n "picture": "",\n "email": "<email>",\n "email_verified": true,\n "locale": "en",\n "hd": "<domain>"\n}\n
[Sat Aug 05 18:58:19.973727 2017] [auth_openidc:debug] [pid 52464] src/proto.c(1968): [client 73.158.230.14:63889] oidc_user_info_response_validate: enter: userinfo_signed_response_alg=(null), userinfo_encrypted_response_alg=(null), userinfo_encrypted_response_enc=(null)
[Sat Aug 05 18:58:19.973757 2017] [auth_openidc:debug] [pid 52464] src/util.c(1957): [client 73.158.230.14:63889] oidc_util_create_symmetric_key: key_len=32
[Sat Aug 05 18:58:19.973815 2017] [auth_openidc:debug] [pid 52464] src/proto.c(2067): [client 73.158.230.14:63889] oidc_proto_resolve_composite_claims: enter
[Sat Aug 05 18:58:19.973827 2017] [auth_openidc:debug] [pid 52464] src/proto.c(2198): [client 73.158.230.14:63889] oidc_proto_resolve_userinfo: id_token_sub=<sub>, user_info_sub=<sub>
[Sat Aug 05 18:58:19.973882 2017] [auth_openidc:debug] [pid 52464] src/util.c(1899): [client 73.158.230.14:63889] oidc_util_json_merge: src={"azp":"<azp>","aud":"<aud>","sub":"<sub>","hd":"<domain>","email":"<email>","email_verified":true,"at_hash":"<hash>","nonce":"<nonce>","iss":"https://accounts.google.com","iat":1501973899,"exp":1501977499,"name":"Dylan Foster","picture":"","given_name":"Dylan","family_name":"Foster","locale":"en"}, dst={"sub":"<sub>","name":"Dylan Foster","given_name":"Dylan","family_name":"Foster","picture":"","email":"<email>","email_verified":true,"locale":"en","hd":"<domain>"}
[Sat Aug 05 18:58:19.973915 2017] [auth_openidc:debug] [pid 52464] src/util.c(1910): [client 73.158.230.14:63889] oidc_util_json_merge: result dst={"sub":"<sub>","name":"Dylan Foster","given_name":"Dylan","family_name":"Foster","picture":"","email":"<email>","email_verified":true,"locale":"en","hd":"<domain>","azp":"<azp>","aud":"<aud>","at_hash":"<hash>","nonce":"<nonce>","iss":"https://accounts.google.com","iat":1501973899,"exp":1501977499}
[Sat Aug 05 18:58:19.973935 2017] [auth_openidc:debug] [pid 52464] src/mod_auth_openidc.c(1571): [client 73.158.230.14:63889] oidc_set_request_user: set remote_user to "<user>@accounts.google.com" based on claim: "sub@"
[Sat Aug 05 18:58:19.973987 2017] [auth_openidc:debug] [pid 52464] src/mod_auth_openidc.c(1621): [client 73.158.230.14:63889] oidc_save_in_session: session management disabled: "check_session_iframe" is not set in provider configuration
[Sat Aug 05 18:58:19.973993 2017] [auth_openidc:debug] [pid 52464] src/mod_auth_openidc.c(1043): [client 73.158.230.14:63889] oidc_store_userinfo_claims: enter
[Sat Aug 05 18:58:19.974020 2017] [auth_openidc:debug] [pid 52464] src/mod_auth_openidc.c(1661): [client 73.158.230.14:63889] oidc_save_in_session: provider->session_max_duration = 28800, session_expires=1096955267
[Sat Aug 05 18:58:19.974026 2017] [auth_openidc:debug] [pid 52464] src/mod_auth_openidc.c(902): [client 73.158.230.14:63889] oidc_log_session_expires: session max lifetime: Sun, 06 Aug 2017 06:58:19 GMT (in 28799 secs from now)
[Sat Aug 05 18:58:19.974031 2017] [auth_openidc:debug] [pid 52464] src/util.c(2114): [client 73.158.230.14:63889] oidc_util_hdr_in_get: Host=<host>
[Sat Aug 05 18:58:19.974066 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(595): [client 73.158.230.14:63889] oidc_cache_set: enter: 3a65b767-859e-4d2a-9ea1-13b43bbe3e06 (section=s, len=2724, encrypt=0, ttl(s)=299, type=shm)
[Sat Aug 05 18:58:19.974085 2017] [auth_openidc:debug] [pid 52464] src/cache/common.c(622): [client 73.158.230.14:63889] oidc_cache_set: successfully stored 2724 bytes in shm cache backend for key 3a65b767-859e-4d2a-9ea1-13b43bbe3e06
[Sat Aug 05 18:58:19.974094 2017] [auth_openidc:debug] [pid 52464] src/util.c(2168): [client 73.158.230.14:63889] oidc_util_hdr_err_out_add: Set-Cookie: mod_auth_openidc_session=3a65b767-859e-4d2a-9ea1-13b43bbe3e06; Path=/; Secure; HttpOnly
[Sat Aug 05 18:58:19.974109 2017] [auth_openidc:debug] [pid 52464] src/mod_auth_openidc.c(1891): [client 73.158.230.14:63889] oidc_handle_authorization_response: session created and stored, returning to original URL: <host>, original method: get
[Sat Aug 05 18:58:19.974114 2017] [auth_openidc:debug] [pid 52464] src/util.c(2146): [client 73.158.230.14:63889] oidc_util_hdr_table_set: Location: <host>
```

**Comment by dylanfoster** [comment]
^ that's the last 50 logs after hitting the protected site

**Comment by dylanfoster** [comment]
```
[Sat Aug 05 19:20:59.602164 2017] [auth_openidc:debug] [pid 52664] src/proto.c(2198): [client 73.158.230.14:51432] oidc_proto_resolve_userinfo: id_token_sub=<sub>, user_info_sub=<sub>, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602204 2017] [auth_openidc:debug] [pid 52664] src/util.c(1899): [client 73.158.230.14:51432] oidc_util_json_merge: src={"azp":"<azp>","aud":"<aud>","sub":"<sub>","hd":"<domain>","email":"<email>","email_verified":true,"at_hash":"<hash>,"nonce":"<nonce>","iss":"https://accounts.google.com","iat":1501975259,"exp":1501978859,"name":"Dylan Foster","picture":"","given_name":"Dylan","family_name":"Foster","locale":"en"}, dst={"sub":"<sub>","name":"Dylan Foster","given_name":"Dylan","family_name":"Foster","picture":"","email":"<email>","email_verified":true,"locale":"en","hd":"<domain>"}, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602231 2017] [auth_openidc:debug] [pid 52664] src/util.c(1910): [client 73.158.230.14:51432] oidc_util_json_merge: result dst={"sub":"<sub>","name":"Dylan Foster","given_name":"Dylan","family_name":"Foster","picture":"","email":"<email>","email_verified":true,"locale":"en","hd":"<domain>","azp":"<azp>","aud":"<aud>","at_hash":<hash>","nonce":"<nonce>","iss":"https://accounts.google.com","iat":1501975259,"exp":1501978859}, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602251 2017] [auth_openidc:debug] [pid 52664] src/mod_auth_openidc.c(1571): [client 73.158.230.14:51432] oidc_set_request_user: set remote_user to "<user>@accounts.google.com" based on claim: "sub@", referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602308 2017] [auth_openidc:debug] [pid 52664] src/mod_auth_openidc.c(1621): [client 73.158.230.14:51432] oidc_save_in_session: session management disabled: "check_session_iframe" is not set in provider configuration, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602318 2017] [auth_openidc:debug] [pid 52664] src/mod_auth_openidc.c(1043): [client 73.158.230.14:51432] oidc_store_userinfo_claims: enter, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602347 2017] [auth_openidc:debug] [pid 52664] src/mod_auth_openidc.c(1661): [client 73.158.230.14:51432] oidc_save_in_session: provider->session_max_duration = 28800, session_expires=-1838383702, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602357 2017] [auth_openidc:debug] [pid 52664] src/mod_auth_openidc.c(902): [client 73.158.230.14:51432] oidc_log_session_expires: session max lifetime: Sun, 06 Aug 2017 07:20:59 GMT (in 28799 secs from now), referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602368 2017] [auth_openidc:debug] [pid 52664] src/util.c(2114): [client 73.158.230.14:51432] oidc_util_hdr_in_get: Host=<host>, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=<client_id>&as=<as>&destination=<site_root>&approval_state=<state>=<state>&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602433 2017] [auth_openidc:debug] [pid 52664] src/cache/common.c(595): [client 73.158.230.14:51432] oidc_cache_set: enter: 7e0853f0-ae3f-4aef-bd1d-77da8c4943fc (section=s, len=2724, encrypt=0, ttl(s)=299, type=shm), referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602455 2017] [auth_openidc:debug] [pid 52664] src/cache/common.c(622): [client 73.158.230.14:51432] oidc_cache_set: successfully stored 2724 bytes in shm cache backend for key 7e0853f0-ae3f-4aef-bd1d-77da8c4943fc, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602467 2017] [auth_openidc:debug] [pid 52664] src/util.c(2168): [client 73.158.230.14:51432] oidc_util_hdr_err_out_add: Set-Cookie: mod_auth_openidc_session=7e0853f0-ae3f-4aef-bd1d-77da8c4943fc; Path=/; Secure; HttpOnly, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602483 2017] [auth_openidc:debug] [pid 52664] src/mod_auth_openidc.c(1891): [client 73.158.230.14:51432] oidc_handle_authorization_response: session created and stored, returning to original URL: <site_root>/, original method: get, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.602490 2017] [auth_openidc:debug] [pid 52664] src/util.c(2146): [client 73.158.230.14:51432] oidc_util_hdr_table_set: Location: <site_root>/, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.709308 2017] [auth_openidc:debug] [pid 52664] src/mod_auth_openidc.c(3302): [client 73.158.230.14:51432] oidc_check_user_id: incoming request: "/?(null)", ap_is_initial_req(r)=1, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.709333 2017] [auth_openidc:debug] [pid 52664] src/util.c(2114): [client 73.158.230.14:51432] oidc_util_hdr_in_get: Cookie=mod_auth_openidc_session=7e0853f0-ae3f-4aef-bd1d-77da8c4943fc; JSESSIONID.9ff7b7e0=node09zvdnwytrf8omf4fvc0kesam8.node0; jenkins-timestamper-offset=25200000, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.709342 2017] [auth_openidc:debug] [pid 52664] src/util.c(980): [client 73.158.230.14:51432] oidc_util_get_cookie: returning "mod_auth_openidc_session" = "7e0853f0-ae3f-4aef-bd1d-77da8c4943fc", referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.709349 2017] [auth_openidc:debug] [pid 52664] src/cache/common.c(536): [client 73.158.230.14:51432] oidc_cache_get: enter: 7e0853f0-ae3f-4aef-bd1d-77da8c4943fc (section=s, decrypt=0, type=shm), referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.709362 2017] [auth_openidc:debug] [pid 52664] src/cache/common.c(570): [client 73.158.230.14:51432] oidc_cache_get: cache hit: return 2724 bytes from shm cache backend for key 7e0853f0-ae3f-4aef-bd1d-77da8c4943fc, referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
[Sat Aug 05 19:20:59.709473 2017] [auth_openidc:debug] [pid 52664] src/util.c(1132): [client 73.158.230.14:51432] oidc_util_request_matches_url: comparing "/"=="/oauth_callback.html", referer: https://accounts.google.com/signin/oauth/oauthchooseaccount?client_id=45685778335-0tr0iq2nn1hni7l7t08hfi29g4uas9qq.apps.googleusercontent.com&as=497266c73a93638c&destination=<site_root>&approval_state=!ChRQSEZ2aWg4VXpNbzRQZWJPbTB4chIfUTJ5MDZlemQwQUVVb1Bud0gtVkU1UUZXY3pkTDJ4VQ%E2%88%99ADiIGyEAAAAAWYekTp8nu9n5TRneQpaXEFvXjIjNtkIp&xsrfsig=AHgIfE9bph-DDSXFdoDBUbB1hpSmxbZMpA&flowName=GeneralOAuthFlow
```

**Comment by zandbelt** [comment]
The conversion of the expiry timestamp to a string seems wrong:
> ... oidc_save_in_session: provider->session_max_duration = 28800, session_expires=-1838383702 ...

and since that timestamp is stored as a string in the session indeed this is where the problem is. I have seen this once before with a self-built module but was not able to find the root cause back then.

It has to do with the `APR_TIME_T_FMT` print modifier not working correctly. It may be a compiler or library version issue. I'm using MacPorts myself and have no problem. What's your output of `uname -a`?
```
Darwin macbook.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
```

**Comment by dylanfoster** [comment]
```
Darwin administrators-Mac-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
```

**Comment by zandbelt** [comment]
What about the compiler environment?
```
gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
and
```
cat /usr/local/include/apr.h | grep -C 15 version
typedef  unsigned short  apr_uint16_t;

typedef  int             apr_int32_t;
typedef  unsigned int    apr_uint32_t;

#define APR_SIZEOF_VOIDP 8

/*
 * Darwin 10's default compiler (gcc42) builds for both 64 and
 * 32 bit architectures unless specifically told not to.
 * In those cases, we need to override types depending on how
 * we're being built at compile time.
 * NOTE: This is an ugly work-around for Darwin's
 * concept of universal binaries, a single package
 * (executable, lib, etc...) which contains both 32
 * and 64 bit versions. The issue is that if APR is
 * built universally, if something else is compiled
 * against it, some bit sizes will depend on whether
 * it is 32 or 64 bit. This is determined by the __LP64__
 * flag. Since we need to support both, we have to
 * handle OS X unqiuely.
 */
#ifdef DARWIN_10
#undef APR_SIZEOF_VOIDP
#undef INT64_C
#undef UINT64_C
#ifdef __LP64__
 typedef  long            apr_int64_t;
 typedef  unsigned long   apr_uint64_t;
 #define APR_SIZEOF_VOIDP     8
 #define INT64_C(v)   (v ## L)
```

**Comment by zandbelt** [comment]
Moreover, can you run `make test` and paste the output of `./test/test-cmd timestamp 300`?

**Comment by zandbelt** [comment]
this was fixed in https://github.com/zmartzone/mod_auth_openidc/commit/99ccd3ca64eb1ea69054c011a95dac16ff16d6cc and included in 2.4.8.2 and upwards
